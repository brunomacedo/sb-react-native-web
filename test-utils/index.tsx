import React, { ReactElement } from "react";
import { render } from "@testing-library/react-native";
import { ThemeProvider } from "../src/theme/ThemeProvider";

const AllTheProviders = ({ children }: { children: ReactElement }) => {
  return (
    <ThemeProvider>
      {children}
    </ThemeProvider>
  )
}

const customRender = (component: ReactElement, options = {}) => (
  render(component, {
    wrapper: AllTheProviders, ...options
  })
);

// re-export everything
export * from "@testing-library/react-native"

// override render method
export { customRender as render }
