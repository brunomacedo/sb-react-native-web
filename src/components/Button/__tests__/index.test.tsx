import React from 'react';
import { fireEvent, render } from '../../../../test-utils';
import { Button } from '../';

describe('Test Button component', () => {
  it('Test click event', () => {
    const mockCallBack = jest.fn();

    const { getByText, toJSON } = render(
      <Button label='Test' onPress={mockCallBack} />
    );

    const button = getByText('Test');
    fireEvent.press(button);
    expect(mockCallBack.mock.calls.length).toEqual(1);
    expect(toJSON()).toMatchSnapshot();
  });
});
