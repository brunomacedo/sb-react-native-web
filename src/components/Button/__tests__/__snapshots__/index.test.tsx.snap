// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Button component Test click event 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
    }
  }
>
  <View
    style={
      [
        {
          "alignItems": "center",
          "borderRadius": 4,
          "flexDirection": "row",
          "justifyContent": "center",
          "minHeight": 48,
          "width": "100%",
        },
        [
          {
            "backgroundColor": "#00B56B",
          },
        ],
      ]
    }
  >
    <Text
      style={
        [
          {
            "color": "#FFFFFF",
            "fontFamily": "CaustenRound-Bold",
            "fontSize": 16,
            "lineHeight": 20,
          },
        ]
      }
    >
      Test
    </Text>
  </View>
</View>
`;
